### PR TITLE
Task/add toss4 dir

### PR DESCRIPTION
**Pull request by kab163**
Adding spack compilers and packages for toss4

### CODE REVIEW

File: toss_4_x86_64_ib_cray/packages.yaml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    externals:
+    - spec: cmake
+      prefix: /usr/tce/packages/cmake/cmake-3.14.5
+  cuda:
+    version: [11.4.120]
+    buildable: false
+
+    externals:
+    - spec: cuda@11.4.120
+      prefix: /opt/toss/cudatoolkit/11.4/
```
Comment (kab163) [inline toss_4_x86_64_ib_cray/packages.yaml:20]: Is this path ok? /opt/toss/cudatoolkit/11.4/bin/nvcc is a valid path, it just looks very different from the other cuda prefix paths..

File: toss_4_x86_64_ib_cray/packages.yaml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    externals:
+    - spec: rocm-device-libs@4.2.0
+      prefix: /opt/rocm-4.2.0/
+    - spec: rocm-device-libs@4.3.1
+      prefix: /opt/rocm-4.3.1/
+    - spec: rocm-device-libs@4.5.0
+      prefix: /opt/rocm-4.5.0/
+    - spec: rocm-device-libs@4.5.2
+      prefix: /opt/rocm-4.5.2/
+  cray_mpich:
```
Comment (kab163) [inline toss_4_x86_64_ib_cray/packages.yaml:81]: Wasn't sure what to put here - what do you think?

File: toss_4_x86_64_ib_cray/packages.yaml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+      prefix: /usr/tce/packages/cmake/cmake-3.14.5
+  cuda:
+    version: [11.4.120]
+    buildable: false
+
+    externals:
+    - spec: cuda@11.4.120
+      prefix: /opt/toss/cudatoolkit/11.4/ 
+  hip:
+    version: [4.2.0, 4.3.1, 4.5.0, 4.5.2]
```
Comment (kab163) [inline toss_4_x86_64_ib_cray/packages.yaml:22]: Do I need all these versions?

**Review by kab163 — COMMENTED** [review]
I have doubts :o

File: toss_4_x86_64_ib_cray/packages.yaml (hunk 3 above)
Comment (davidbeckingsale) [reply toss_4_x86_64_ib_cray/packages.yaml:22]: I would only include the available versions - if they are all on the machine it's fine to include them.

File: toss_4_x86_64_ib_cray/packages.yaml (hunk 1 above)
Comment (davidbeckingsale) [reply toss_4_x86_64_ib_cray/packages.yaml:20]: Sure - I'm not ever sure people will be using it.

File: toss_4_x86_64_ib_cray/packages.yaml (hunk 2 above)
Comment (davidbeckingsale) [reply toss_4_x86_64_ib_cray/packages.yaml:81]: Good question - I think this is okay for now and we can add more MPI at a later time.

File: toss_4_x86_64_ib_cray/packages.yaml (hunk 3 above)
Comment (kab163) [reply toss_4_x86_64_ib_cray/packages.yaml:22]: They are all available in the /opt/rocm-4.*/ dirs, so I'll leave it for now!

**Pull request merged**